### PR TITLE
[None][perf] GVR V2 top-K: 4K<n<=8K register rungs, unified QC gate, sampled small-envelope prefill plan, SM-count-aware dispatch with B300 tuning

### PR DESCRIPTION
**Pull request by longcheng-nv**
## Description

This PR updates the host-side dispatch for self-sampling GVR V2 top-K. It improves decode and prefill routing on B200/B300; the device kernel is unchanged.

The existing dispatch has four issues:

- `4K < n <= 8K` decode shapes miss faster register-resident plans;
- the large crossing-bin gate used by some register plans can make one difficult row stall the launch;
- small-row prefill can select an unsampled plan inside the range where sampling is faster;
- register-wave boundaries assume a fixed 148-SM device.

## Changes

- **Decode:** add register routes for `4K < n <= 8K`:
  - `b <= num_sms`: `BLK=1024`, `VPT=2`;
  - `num_sms < b <= 2 * num_sms`: `BLK=512`, `VPT=4`, `MINB=2`.
- **Crossing-bin gate:** use `QC = QUADC = 96` for all register plans.
- **Prefill:** for at most 148 rows, use the sampled `BLK=512` plan when `n_env` is above its sampling threshold plus 256 and no larger than 16K. `run_prefill`, `prefill_ready`, and `warmup_prefill` now share this rule.
- **Device-aware routing:** read and cache each CUDA device's SM count and compute capability, and include that profile in launcher/warmup cache keys.
- **B300:** on SM103 with 148 SMs, route the measured `b=512..1024`, `k=2048`, `4K < n <= 8K` shapes to the existing `BLK=512` register plan. SM100 routing is unchanged by this exception.

Public routing helpers keep their existing behavior through default arguments. `prefill_ready` adds an optional `max_row_len` so callers can use the same envelope as `run_prefill`.

## Performance

Ratios are `main / PR`; values above 1 mean the PR is faster.

| Device / population | Cases | Result |
| --- | ---: | --- |
| B200 changed decode routes | 1,008 | **1.153x** geometric mean; **1.067x** p5 |
| B200 changed prefill routes | 153 | **2.04x** geometric mean; **1.51x** minimum |
| B300 changed SM103 routes | 122 | **+9.22% cold**, **+18.49% warm** |
| Unchanged B200/B300 routes | 19,670+ | Neutral within measurement noise |

The unified QC gate reduces the targeted real-row case from `19.0 us` to `6.1 us`.

## Test Coverage

- Tie-aware exactness against `torch.topk`:
  - B200 decode: `19,492 / 19,492` arm-cases;
  - B200 prefill: `4,380 / 4,380` arm-cases;
  - reciprocal B200/B300 validation: exact in every measured arm-case.
- `tests/unittest/_torch/thop/parallel/test_gvr_selfsampling_topk.py`: **123 tests passed on B200**.
- Tests cover route boundaries, SM scaling and cache isolation, varlen exactness, MTP windows, CUDA graph replay, and prefill warmup/readiness.
- Standard CI passed.

## Reviewer Notes

The PR changes two files:

1. `tensorrt_llm/_torch/cute_dsl_kernels/blackwell/top_k/gvr_topk_decode_self_sampling_host.py`: dispatch rules, device-profile cache keys, and prefill lifecycle consistency.
2. `tests/unittest/_torch/thop/parallel/test_gvr_selfsampling_topk.py`: route, correctness, MTP, CUDA graph, and cache-isolation coverage.

Out of scope: device-kernel changes, longer-context live-length routing, and additional prefill tuning outside the validated window.

## PR Checklist

- [x] PR description explains what and why.
- [x] New dispatch paths have test coverage.
- [x] DCO sign-off and local pre-commit requirements are satisfied.

### CODE REVIEW

File: tests/unittest/_torch/thop/parallel/test_gvr_selfsampling_topk.py [hunk 1]
```diff
@@ -878,6 +923,113 @@ def test_selfsampling_varlen_reg_cuda_graph():
...
+    ss_host.run_varlen(logits, kv_lens, out, next_n=1, compress_ratio=1, max_seq_len=n)
+    torch.cuda.synchronize()
+    key = (rows, npad, top_k, n, 1, 1)
+    assert ss_host._VARLEN_CACHE[key][0] == "reg", ss_host._VARLEN_CACHE[key][0]
+    ref = _reference_varlen_indices(logits, kv_lens, 1, 1, top_k)
+    for r in range(rows):
+        row = logits[r]
+        got = row[out[r].long().clamp_min(0)].sort().values
+        want_v = row[ref[r].long().clamp_min(0)].sort().values
+        assert torch.equal(got, want_v), f"row {r} value multiset mismatch"
```
[the remark this answers is not in the document]
Comment (longcheng-nv) [reply tests/unittest/_torch/thop/parallel/test_gvr_selfsampling_topk.py:1207]: Done in 54e28982e8: the head of every row is now asserted unique (the two-valued row 3 is the case where duplicates could hide behind the value multiset).

**Review by lfr-0531 — APPROVED** [review]

**Review by zongfeijing — APPROVED** [review]
